README.md

```diff
@@ -1,3 +1,227 @@
+
 # odjitter
 
-Don't rely on anything here yet, API in flux...
+This crate contains an implementation of the ‘jittering’ technique for
+pre-processing origin-destination (OD) data. Jittering in a data
+visualisation context refers to the addition of “random noise to the
+data” to prevent points in graphs from overlapping, as described in by
+Wickham et al. ([2016](https://ggplot2-book.org/)) and in the
+documentation page for the function
+[`geom_jitter()`](https://ggplot2.tidyverse.org/reference/geom_jitter.html).
+
+In the context of OD data jittering refers to randomly moving start and
+end points associated with OD pairs, as described in an under review
+paper on the subject (Lovelace et al. under review). The technique is
+implemented in the function
+[`od_jitter()`](https://itsleeds.github.io/od/reference/od_jitter.html)
+in the [`od`](https://itsleeds.github.io/od/index.html) R package. The
+functionality contained in this repo is an extended and much faster
+implementation: according to our benchmarks on a large dataset it was
+around 1000 times faster than the R implementation.
+
+The crate is still a work in progress: the API may change. Issues and
+pull requests are particularly useful at this stage.
+
+<!-- Todo: add worked example. Will do for GISRUK paper -->
+<!-- An illustration of its functionality is shown below. -->
+
+# Installation
+
+Install the package from the system command line as follows (you need to
+have installed and set-up
+[cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html)
+first):
+
+``` bash
+cargo install --git https://github.com/dabreegster/odjitter
+```
+
+To check the package installation worked, you can run `odjitter` command
+without arguments. If it prints the following message congratulations,
+it works 🎉
+
+``` bash
+odjitter
+```
+
+    ## error: The following required arguments were not provided:
+    ##     --od-csv-path <OD_CSV_PATH>
+    ##     --zones-path <ZONES_PATH>
+    ##     --output-path <OUTPUT_PATH>
+    ##     --max-per-od <MAX_PER_OD>
+    ## 
+    ## USAGE:
+    ##     odjitter [OPTIONS] --od-csv-path <OD_CSV_PATH> --zones-path <ZONES_PATH> --output-path <OUTPUT_PATH> --max-per-od <MAX_PER_OD>
+    ## 
+    ## For more information try --help
+
+# Usage
+
+To run algorithm you need a minimum of three inputs, examples of which
+are provided in the `data/` folder of this repo:
+
+1.  A .csv file containing OD data with the first two columns containing
```

Comment:
Order of columns doesn't matter. The CLI has `--origin-key` and `--destination-key` for this reason. I am philosophically weirded out by the concept of column order mattering in CSV; it's an unordered mapping from keys to values, isn't it?

Reply:
Yes column order shouldn't matter. Happy to change that. But how should people specify which columns have the IDs in?

Reply:
By saying `--origin-key=from --destination-key=to` or whatever in the CLI. By default, the CLI defaults to geo_code1 and 2

Reply:
Yep, that's important and worth mentioning. Will update...